src/main/kotlin/org/rust/ide/annotator/RsErrorAnnotator.kt

```diff
@@ -82,6 +82,35 @@ class RsErrorAnnotator : RsAnnotatorBase(), HighlightRangeExtension {
         element.accept(visitor)
     }
 
+    private fun checkEnumItem(holder: AnnotationHolder, o: RsEnumItem) {
+        checkDuplicates(holder, o)
+        o.enumBody?.let { checkDuplicateEnumVariants(holder, it) }
+    }
+
+    private fun checkDuplicateEnumVariants(holder: AnnotationHolder, o: RsEnumBody) {
+        data class VariantInfo(val variant: RsEnumVariant, val alreadyReported: Boolean)
+
+        var discrCounter = 0
+        val indexToVariantMap = hashMapOf<Int, VariantInfo>()
+        for (variant in o.enumVariantList) {
+            val literal = variant.variantDiscriminant?.expr as? RsLitExpr
+            val int = literal?.integerLiteralValue?.toIntOrNull()
```

Comment:
Should we use Long instead?

Reply:
Will be fixed in #3796

Reply:
Also, it will automatically support binary, octal and hex literals with a new API for literal values

Reply:
@Undin Actually, won't BigInteger be better? With `#![feature(repr128)]` you can have the discriminant be over 64 bits long. I'm not sure if that's justification enough though?